presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt

```diff
@@ -87,110 +81,219 @@ fun EventScreen(
+                                fontWeight = FontWeight.Medium,
+                                fontSize = 16.sp,
+                                color = MaterialTheme.colorScheme.visitorTextFontColor
+                            )
+                            Spacer(modifier = Modifier.height(16.dp))
                         }
-                    )
 
-                    Spacer(modifier = modifier.height(26.dp))
+                        if(eventScreenState.filteredVisitorsNotGoing.isNotEmpty()) {
+                            items(eventScreenState.filteredVisitorsNotGoing) { attendee ->
+                                VisitorItem(
+                                    initials = attendee.fullName.toInitials(),
+                                    userId = attendee.userId,
+                                    fullName = attendee.fullName,
+                                    isCreator = eventScreenState.host == attendee.userId,
+                                    onDeleteClicked = { userId ->
+                                        eventScreenEvent(EventScreenEvent.OnDeleteVisitor(userId))
+                                    }
+                                )
+                            }
+                        }
+                    }
 
-                    AgendaAction(
-                        agendaActionType = if(eventScreenState.isUserEventCreator) { AgendaActionType.DELETE_EVENT } else { AgendaActionType.JOIN_EVENT },
-                        onActionClicked = { agendaActionType ->
-                            when(agendaActionType) {
-                                AgendaActionType.DELETE_EVENT -> {
-                                    eventScreenEvent(EventScreenEvent.OnShowDeleteEventAlertDialog(shouldShowDeleteAlertDialog = true))
-                                }
-                                AgendaActionType.JOIN_EVENT -> {
+                    item {
+                        Spacer(modifier = modifier.height(26.dp))
+                        AgendaAction(
+                            agendaActionType = if (eventScreenState.isUserEventCreator) {
+                                AgendaActionType.DELETE_EVENT
+                            } else {
+                                /** Based on if the attendee has updated isGoing status
+                                 * isGoing = true ==> LEAVE_EVENT
+                                 * isGoing = false ==> JOIN_EVENT
+                                 * */
 
-                                }
-                                AgendaActionType.LEAVE_EVENT -> {
+                                /* Find the attendee and check their status - TODO Consider a helper method for this */
+                                val isAttendeeGoing = eventScreenState.attendees.firstOrNull { attendee ->
+                                    attendee.userId == eventScreenState.currentLoggedInUserId
+                                }?.isGoing ?: false
```

Comment:
I would have this on the viewmodel. Try to keep the UI as dumb as possible, and not make much calculations

Again, this won't take much performance, but if in the future you add something complex, it's better for you to get used to adding it directly on the ViewModel
This is due to recomposition

Reply:
@MKiperszmid Understand, will move this to the ViewModel